string/regex matcher/8449789e-3b23-4bf5-9c33-931015049dc5.json

```diff
@@ -1,28 +1,38 @@
 {
-  "description": "Executes a global regex pattern match.\n\nFlavour depends on the running environment.",
+  "description": "Matches a string to a non-global regular expression, and returns substrings that match the regular expression's groups.\n\n",
   "interface": {
     "inputs": {
       "e79c5f02-8a3c-42e7-95cc-79bfe07a4784": {
         "type": "string",
         "name": "string",
-        "description": "The string."
+        "description": "Receives the string to be matched.\r\n\r\nExample:\r\n\r\n\"The quick brown fox jumps over the lazy dog. It barked.\""
       },
       "4ad33973-93b5-42cd-b485-76c517507786": {
         "type": "string",
         "name": "regex",
-        "description": "The regex pattern"
+        "description": "Receives the regular expression.\r\n\r\nExample:\r\n\r\n\"(?<animal>fox|cat) jumps over\""
       }
     },
     "outputs": {
       "b3b69745-686f-4978-a5cf-ef61c31da582": {
         "type": "boolean",
         "name": "matches",
-        "description": "Whether the given regex matches the given string"
+        "description": "Whether the given regex matches the given string;\n\nExample: \n\ntrue"
+      },
+      "8619dc0b-682c-475e-9820-b0a9a5274eed": {
+        "type": "string",
+        "name": "matched",
+        "description": "Sends the matched text.\n\nExample: \n\n\"fox jumps over\""
       },
       "0ae97343-7bd6-43c7-bc0a-08acc2b964c8": {
         "type": "string[]",
         "name": "groups",
-        "description": "Groups matched by the given regex pattern"
+        "description": "Sends the capturing groups matched by the given regex pattern.\n\nExample:\n```\n[ \"fox\" ]\n```"
+      },
+      "856b46c6-6d81-4023-989e-3e312f051e02": {
+        "type": "({string: string} or null)",
```

Comment:
I think we decided to emit empty object instead of null it there weren't any named groups, didn't we?

Reply:
You are right.